public/video-ui/src/components/VideoUpload/YoutubeUpload.js

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import Icon from '../Icon';
+import { ManagedForm } from '../ManagedForm';
+import VideoUtils from '../../util/video';
+
+export default class YoutubeUpload extends React.Component {
+  state = { file: null };
+
+  setFile = event => {
+    if (!this.props.video) {
+      return;
+    }
+
+    if (event.target.files.length === 0) {
+      this.setState({ file: null });
+    } else {
+      this.setState({ file: event.target.files[0] });
+    }
+  };
+
+  render() {
+    const { video, startUpload } = this.props;
+
+    const canUploadToYouTube = VideoUtils.canUploadToYouTube(video);
+
+    return (
+      <div className="video__detailbox video__detailbox__assets">
+        <div className="video__detailbox__header__container">
+          <header className="video__detailbox__header">
+            Upload to YouTube
+          </header>
+        </div>
+        <div className="form__group">
+          <ManagedForm
+            data={video}
+            updateData={this.props.saveVideo}
+            editable={true}
+            formName="YouTubeUpload"
+          >
+            <input
+              className="form__field"
+              type="file"
+              onChange={this.setFile}
+              disabled={!canUploadToYouTube || this.props.uploading}
```

Comment:
if they cannot upload to youtube because of missing data, should we tell them why - it could be confusing because teh missing fields are on the main video page.

Reply:
👍 

![image](https://user-images.githubusercontent.com/836140/33704705-616d8878-db25-11e7-8b5f-2de8a71773ee.png)